bip-xxxx.mediawiki

```diff
@@ -0,0 +1,111 @@
+<pre>
+  BIP: ?
+  Layer: Applications
+  Title: unspendable() Descriptor Key Expression
+  Author: Andrew Toth <andrewstoth@gmail.com>
+          Kewde <kewde@particl.io>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-?
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-12-12
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This document specifies a <tt>unspendable()</tt> key expression for output script descriptors. The <tt>unspendable()</tt> expression operates on the root <tt>TREE</tt> expression and produces an unspendable public key that can be independently verified by anyone with knowledge of all the constituent public keys.
+
+==Copyright==
+
+This BIP is licensed under the BSD 2-clause license.
+
+==Motivation==
+
+When creating a multi-party Taproot transaction spending only from the script path, it is useful to be able to prove to all cosigners that they keypath is unspendable. Otherwise a malicious participant could use an internal key which they have the private key for and spend the transaction out from the rest of the participants.
+
+This document introduces a mechanism to compute a NUMS (Nothing Up My Sleeve) point for use in the Taproot key path that:
+* Allows active participants involved in constructing the output script to independently verify the unspendable key.
+* Prevents passive observers from recognizing that the key path is unspendable.
+* Enables signers with limited information, such as hardware wallets, to verify unspendability without requiring user interaction.
+
+==Specification==
+
+A new key expression is defined: <tt>unspendable()/NUM/.../*</tt>.
+
+===<tt>unspendable()/NUM/.../*</tt>===
+
+The <tt>unspendable</tt> expression can only be used as the first argument of a BIP386 <tt>tr(KEY, TREE)</tt> expression. All other <tt>KEY</tt> expressions in the descriptor must be <tt>xpub</tt> encoded extended public keys with exactly 2 unhardened derivation steps. The derivation steps may include <tt>/*</tt> or a BIP389 multipath expression, but still must have only unhardened steps. BIP390 <tt>musig(KEY, KEY, ..., KEY)</tt> expressions are allowed, but the variant with derivation after the expression <tt>musig(KEY, KEY, ..., KEY)/NUM/.../*</tt> is forbidden.
```

Comment:
I don't understand this limitation over `musig`.
`musig(KEY, KEY, ..., KEY)/<M,N>/*` is the scheme supported for MuSig2 in wallet policies.

Reply:
If `musig(KEY, KEY, ..., KEY)` resolves to `AGG_KEY`, then `tr(unspendable()/0,musig(KEY, KEY, ..., KEY)/<M,N>/*))` and `tr(unspendable()/0,AGG_KEY/<M,N>/*))` would both resolve to the same merkle root but different internal keys. If using just `tr(unspendable()/0,musig(KEY, KEY, ..., KEY)))`, the corresponding `tr(unspendable()/0,AGG_KEY))` is invalid since `AGG_KEY` is not an xpub with 2 derivation paths.

Reply:
One of the goals is also to not have any malleability - two descriptors or policies that produce the same merkle root should always produce the same internal key.
We cannot know if an xpub in a descriptor or wallet policy is actually an aggregate xpub, which could also be represented with the `musig` expression.

Reply:
> replace all xpub/keypath expression with their hex pubkey

The proposal here though prohibits any key expression that is not an xpub with exactly unhardened depth of 2.